sources/Tools/Visualiser.py

```diff
@@ -196,17 +200,25 @@ def _exit(exception=None, message="Error", mod=-1):
         logging.error(f"{message}")
         sys.exit(mod)
 
+    @property
+    def _get_houses(self):
+        return list(set(self.raw_data.loc[:, "Hogwarts House"]))
+
     def __init__(self, func=_scatter_plot_visualizer):
         # could add different matplotlib backend | for now to much work
         super().__init__()
-        self.visualizer_func = self.get_args("type_visualizer", default_value=func)
+        # check this setter for every program usable to prevent crash
+        for k, v in self.get_args("type_visualizer", default_value={"advanced": func}).items():
+            self.header_visualizer = k
+            self.func_visualizer = v
 
     """
         Public methods
     """
 
     def visualizer(self, header):
-        matplotlib.use("pdf")
+        # matplotlib.use("pdf")

```

Comment:
the allow the pdf backend, why did you comment it?

Reply:
the time to test with plt.show()